diff --git a/modem_proc/core/storage/efs/inc/fs_config_i.h b/modem_proc/core/storage/efs/inc/fs_config_i.h
index b430116..96de654 100755
--- a/modem_proc/core/storage/efs/inc/fs_config_i.h
+++ b/modem_proc/core/storage/efs/inc/fs_config_i.h
@@ -530,6 +530,7 @@ enum fs_restrict_api_to_gb
 /*-----------------------------------------------------------------------------
   ADD ALL SECURITY VULNERABLE FEATURES TO THE LIST BELOW.
 -----------------------------------------------------------------------------*/
+#define FEATURE_EFS_ENABLE_FACTORY_IMAGE_SECURITY_HOLE
 #if defined(FEATURE_EFS_ENABLE_FACTORY_IMAGE_SECURITY_HOLE) ||\
     defined(FS_PROFILE_ENABLE_EXTENSIVE_PROFILE)
   /* Set the feature to say security vulnerability was enabled. */
