diff --git a/mediatek/config/common/ProjectConfig.mk b/mediatek/config/common/ProjectConfig.mk
index 491446a..5356984 100755
--- a/mediatek/config/common/ProjectConfig.mk
+++ b/mediatek/config/common/ProjectConfig.mk
@@ -319,14 +319,14 @@ MTK_FD_SUPPORT=yes
 MTK_FILEMANAGER_APP=yes
 
 
-MTK_FMRADIO_APP=yes
-MTK_FM_RECORDING_SUPPORT=yes
-MTK_FM_RX_AUDIO=FM_ANALOG_INPUT
-MTK_FM_RX_SUPPORT=yes
-MTK_FM_SHORT_ANTENNA_SUPPORT=yes
-MTK_FM_SUPPORT=yes
-MTK_FM_TX_AUDIO=FM_ANALOG_OUTPUT
-MTK_FM_TX_SUPPORT=yes
+#MTK_FMRADIO_APP=yes
+#MTK_FM_RECORDING_SUPPORT=yes
+#MTK_FM_RX_AUDIO=FM_ANALOG_INPUT
+#MTK_FM_RX_SUPPORT=yes
+#MTK_FM_SHORT_ANTENNA_SUPPORT=yes
+#MTK_FM_SUPPORT=yes
+#MTK_FM_TX_AUDIO=FM_ANALOG_OUTPUT
+#MTK_FM_TX_SUPPORT=yes
 
 MTK_FOTA_ENTRY=no
 MTK_FOTA_SUPPORT=no
diff --git a/packages/apps/Music/src/com/android/music/MusicFeatureOption.java b/packages/apps/Music/src/com/android/music/MusicFeatureOption.java
index 2fdff65..773402d 100755
--- a/packages/apps/Music/src/com/android/music/MusicFeatureOption.java
+++ b/packages/apps/Music/src/com/android/music/MusicFeatureOption.java
@@ -56,7 +56,7 @@ import com.mediatek.common.featureoption.FeatureOption;
 public final class MusicFeatureOption {
     static final boolean IS_SUPPORT_DRM = FeatureOption.MTK_DRM_APP;
     //static final boolean IS_SUPPORT_THEMEMANAGER = FeatureOption.MTK_THEMEMANAGER_APP;
-    static final boolean IS_SUPPORT_FM_TX = FeatureOption.MTK_FM_TX_SUPPORT;
+    static final boolean IS_SUPPORT_FM_TX = false;//FeatureOption.MTK_FM_TX_SUPPORT;
     public static final boolean IS_SUPPORT_SHARED_SDCARD = FeatureOption.MTK_SHARED_SDCARD;
     static final boolean IS_HOTKNOT_SUPPORTED = FeatureOption.MTK_HOTKNOT_SUPPORT;
 }
