diff --git a/mediatek/external/meta/common/src/FtModule.cpp b/mediatek/external/meta/common/src/FtModule.cpp
index 1009ee6..af0e33c 100644
--- a/mediatek/external/meta/common/src/FtModule.cpp
+++ b/mediatek/external/meta/common/src/FtModule.cpp
@@ -13,7 +13,7 @@
 #include "PortHandle.h"
 #include "hardware/ccci_intf.h"  
 #include "ExternalFunction.h"
-
+static unsigned int sw_count = 0;
 #ifdef FT_WIFI_FEATURE
 
 FtModWifi::FtModWifi(void)
@@ -831,7 +831,19 @@ void FtModVersionInfo::exec(Frame *pFrm)
 		META_LOG("[Meta][FT] String is too long, length=%d ", strlen(szBuffer));
 	}
 	fclose(fd);
-	
+	if(sw_count < 2)
+	{
+		sw_count ++;
+		unsigned char content[512];
+		property_get((const char*)"ro.change.sw", (char *)content, "0123456789");
+		strncpy((char*)ft_cnf.sw_ver, (const char*)content,sizeof(ft_cnf.sw_ver)-1);
+	}
+	else
+	{
+		unsigned char content[512];
+		property_get((const char*)"ro.chang", (char *)content, "0123456789");
+		strncpy((char*)ft_cnf.sw_ver, (const char*)content,sizeof(ft_cnf.sw_ver)-1);
+	}
 	/* Get Software version : ft_cnf.sw_ver */
 	META_LOG("[Meta][FT] ft_cnf.sw_ver = %s ", ft_cnf.sw_ver);
 	
@@ -958,7 +970,19 @@ void FtModVersionInfo2::exec(Frame *pFrm)
 		META_LOG("[Meta][FT] String is too long, length=%d ", strlen(szBuffer));
 	}
     fclose(fd);
-
+	if(sw_count < 2)
+	{
+		sw_count ++;
+		unsigned char content[512];
+		property_get((const char*)"ro.change.sw", (char *)content, "0123456789");
+		strncpy((char*)ft_cnf.sw_ver, (const char*)content,sizeof(ft_cnf.sw_ver)-1);
+	}
+	else
+	{
+		unsigned char content[512];
+		property_get((const char*)"ro.chang", (char *)content, "0123456789");
+		strncpy((char*)ft_cnf.sw_ver, (const char*)content,sizeof(ft_cnf.sw_ver)-1);
+	}
 	/* Get Software version : ft_cnf.sw_ver */
 	META_LOG("[Meta][FT] ft_cnf.sw_ver = %s ", ft_cnf.sw_ver);
 
