diff --git a/mediatek/custom/common/kernel/cam_cal/ar1820otp/ar1820otp.c b/mediatek/custom/common/kernel/cam_cal/ar1820otp/ar1820otp.c
index 5efe1b8..4bb85d4 100755
--- a/mediatek/custom/common/kernel/cam_cal/ar1820otp/ar1820otp.c
+++ b/mediatek/custom/common/kernel/cam_cal/ar1820otp/ar1820otp.c
@@ -18,7 +18,7 @@
 #include <asm/system.h>  // for SMP
 
 //#define CAM_CALGETDLT_DEBUG
-#define CAM_CAL_DEBUG
+//#define CAM_CAL_DEBUG
 #ifdef CAM_CAL_DEBUG
 #define CAM_CALDB printk
 #else
diff --git a/mediatek/custom/common/kernel/imgsensor/ar1820_mipi_raw/ar1820mipi_Sensor.c b/mediatek/custom/common/kernel/imgsensor/ar1820_mipi_raw/ar1820mipi_Sensor.c
index 6aceb7b..b352364 100755
--- a/mediatek/custom/common/kernel/imgsensor/ar1820_mipi_raw/ar1820mipi_Sensor.c
+++ b/mediatek/custom/common/kernel/imgsensor/ar1820_mipi_raw/ar1820mipi_Sensor.c
@@ -42,7 +42,7 @@
 #define LOG_2 LOG_INF("preview 2456*1842@30fps,533.33Mbps/lane; video 3840*2160@30fps,533.33Mbps/lane; capture 18M@15fps,533.33Mbps/lane\n")
 /****************************   Modify end    *******************************************/
 
-#define LOG_INF(format, args...)    xlog_printk(ANDROID_LOG_INFO   , PFX, "[%s] " format, __FUNCTION__, ##args)
+#define LOG_INF(format, args...)
 
 static DEFINE_SPINLOCK(imgsensor_drv_lock);
 
diff --git a/mediatek/custom/common/kernel/imgsensor/src/kd_sensorlist.c b/mediatek/custom/common/kernel/imgsensor/src/kd_sensorlist.c
index 1afb2b4..90ffa3c 100755
--- a/mediatek/custom/common/kernel/imgsensor/src/kd_sensorlist.c
+++ b/mediatek/custom/common/kernel/imgsensor/src/kd_sensorlist.c
@@ -44,7 +44,7 @@ static struct i2c_board_info __initdata i2c_devs1={I2C_BOARD_INFO(CAMERA_HW_DRVN
 #define PK_DBG_FUNC(fmt, arg...)    xlog_printk(ANDROID_LOG_DEBUG , PFX, fmt, ##arg)
 
 
-#define DEBUG_CAMERA_HW_K
+//#define DEBUG_CAMERA_HW_K
 #ifdef DEBUG_CAMERA_HW_K
 #define PK_DBG PK_DBG_FUNC
 #define PK_ERR(fmt, arg...)         xlog_printk(ANDROID_LOG_ERROR , PFX , fmt, ##arg)
diff --git a/mediatek/custom/mt6595/kernel/imgsensor/src/kd_sensorlist.c b/mediatek/custom/mt6595/kernel/imgsensor/src/kd_sensorlist.c
index 5b6d745..b538f3e 100755
--- a/mediatek/custom/mt6595/kernel/imgsensor/src/kd_sensorlist.c
+++ b/mediatek/custom/mt6595/kernel/imgsensor/src/kd_sensorlist.c
@@ -60,7 +60,7 @@ static struct i2c_board_info __initdata i2c_devs2={I2C_BOARD_INFO(CAMERA_HW_DRVN
 #define PK_DBG_FUNC(fmt, arg...)    xlog_printk(ANDROID_LOG_DEBUG , PFX, fmt, ##arg)
 
 
-#define DEBUG_CAMERA_HW_K
+//#define DEBUG_CAMERA_HW_K
 #ifdef DEBUG_CAMERA_HW_K
 #define PK_DBG PK_DBG_FUNC
 #define PK_ERR(fmt, arg...)         xlog_printk(ANDROID_LOG_ERROR , PFX , fmt, ##arg)
diff --git a/mediatek/kernel/drivers/power/battery_common.c b/mediatek/kernel/drivers/power/battery_common.c
index 2b3ee91..be07696 100755
--- a/mediatek/kernel/drivers/power/battery_common.c
+++ b/mediatek/kernel/drivers/power/battery_common.c
@@ -1730,6 +1730,7 @@ PMU_STATUS do_batt_temp_state_machine(void)
 	if (BMT_status.temperature >= g_MAX_CHARGE_TEMPERATURE) 
 	{
 		battery_xlog_printk(BAT_LOG_CRTI, "[BATTERY] Battery Over Temperature !!\n\r"); 
+		printk(KERN_ERR "Battery over Tem %s:[%d]...\n", __func__, __LINE__);
 		g_batt_temp_status =TEMP_POS_HIGH;
 		return PMU_STATUS_FAIL; 
 	}
@@ -1975,6 +1976,7 @@ static PMU_STATUS mt_battery_CheckBatteryTemp(void)
     if (BMT_status.temperature >= g_MAX_CHARGE_TEMPERATURE)
     {
         battery_xlog_printk(BAT_LOG_CRTI, "[BATTERY] Battery Over Temperature !!\n\r");                
+		printk(KERN_ERR "Battery over Tem %s:[%d]...\n", __func__, __LINE__);
         status = PMU_STATUS_FAIL;       
     }    
 #endif
@@ -2165,6 +2167,7 @@ static void mt_battery_notify_VBatTemp_check(void)
 	if(BMT_status.temperature >= g_MAX_CHARGE_TEMPERATURE)
     {
         g_BatteryNotifyCode |= 0x0002;
+		printk(KERN_ERR "Battery over Tem %s:[%d]...\n", __func__, __LINE__);
         battery_xlog_printk(BAT_LOG_CRTI, "[BATTERY] bat_temp(%d) out of range(too high)\n", BMT_status.temperature);		
     }
 #if defined(MTK_JEITA_STANDARD_SUPPORT)
diff --git a/mediatek/kernel/include/linux/hwmsen_helper.h b/mediatek/kernel/include/linux/hwmsen_helper.h
index 575a3c6..2ffd7fa 100755
--- a/mediatek/kernel/include/linux/hwmsen_helper.h
+++ b/mediatek/kernel/include/linux/hwmsen_helper.h
@@ -27,12 +27,20 @@
 /******************************************************************************
  * MACRO & DEFINITION
 ******************************************************************************/ 
+#define HWM_LOG_SWITCH 0
 #define C_I2C_FIFO_SIZE         8       /*according i2c_mt6516.c*/
 #define HWM_TAG					"<HWMSEN> "
+#if HWM_LOG_SWITCH
 #define HWM_FUN(f)				printk(HWM_TAG"%s\n", __func__)
 #define HWM_ERR(fmt, args...)	printk(HWM_TAG"%s %d : "fmt, __func__, __LINE__, ##args)
 #define HWM_LOG(fmt, args...)	printk(HWM_TAG fmt, ##args)
 #define HWM_VER(fmt, args...)   printk(HWM_TAG"%s: "fmt, __func__, ##args) //((void)0)
+#else
+#define HWM_FUN(f)				
+#define HWM_ERR(fmt, args...)	
+#define HWM_LOG(fmt, args...)	
+#define HWM_VER(fmt, args...)   
+#endif
 /******************************************************************************
  * STRUCTURE & ENUMERATION
 ******************************************************************************/ 
diff --git a/mediatek/platform/common/kernel/drivers/cmdq/cmdq_core.c b/mediatek/platform/common/kernel/drivers/cmdq/cmdq_core.c
index 725047a..c27daeb 100755
--- a/mediatek/platform/common/kernel/drivers/cmdq/cmdq_core.c
+++ b/mediatek/platform/common/kernel/drivers/cmdq/cmdq_core.c
@@ -4493,7 +4493,7 @@ int32_t cmdqCoreReleaseTask(TaskStruct *pTask)
     }
 
     cmdq_core_track_task_record(pTask, thread);
-    printk("<3>cmdq_core_release_thread cmdqCoreReleaseTask\n");
+    CMDQ_MSG("<3>cmdq_core_release_thread cmdqCoreReleaseTask\n");
     cmdq_core_release_thread(pTask);
     cmdq_core_release_task(pTask);
     CMDQ_MSG("-->TASK: cmdqCoreReleaseTask 0x%p end\n", pTask);
@@ -4544,7 +4544,7 @@ int32_t cmdqCoreWaitResultAndReleaseTask(TaskStruct *pTask, cmdqRegValueStruct *
         }
         mutex_unlock(&gCmdqTaskMutex);
     }
-    printk("<3>cmdq_core_release_thread cmdqCoreWaitResultAndReleaseTask\n");
+    //printk("<3>cmdq_core_release_thread cmdqCoreWaitResultAndReleaseTask\n");
     cmdq_core_track_task_record(pTask, thread);
     cmdq_core_release_thread(pTask);
     cmdq_core_release_task(pTask);
diff --git a/mediatek/platform/mt6595/hardware/mtkcam/core/drv/isp/isp_drv.cpp b/mediatek/platform/mt6595/hardware/mtkcam/core/drv/isp/isp_drv.cpp
index 9ee4e16..58232bb 100755
--- a/mediatek/platform/mt6595/hardware/mtkcam/core/drv/isp/isp_drv.cpp
+++ b/mediatek/platform/mt6595/hardware/mtkcam/core/drv/isp/isp_drv.cpp
@@ -56,12 +56,22 @@ DECLARE_DBG_LOG_VARIABLE(isp_drv);
 #undef LOG_WRN
 #undef LOG_ERR
 #undef LOG_AST
+#define isp_log_switch 0
+#if isp_log_switch
 #define LOG_VRB(fmt, arg...)        do { if (isp_drv_DbgLogEnable_VERBOSE) { BASE_LOG_VRB(fmt, ##arg); } } while(0)
 #define LOG_DBG(fmt, arg...)        do { if (isp_drv_DbgLogEnable_DEBUG  ) { BASE_LOG_DBG(fmt, ##arg); } } while(0)
 #define LOG_INF(fmt, arg...)        do { if (isp_drv_DbgLogEnable_INFO   ) { BASE_LOG_INF(fmt, ##arg); } } while(0)
 #define LOG_WRN(fmt, arg...)        do { if (isp_drv_DbgLogEnable_WARN   ) { BASE_LOG_WRN(fmt, ##arg); } } while(0)
 #define LOG_ERR(fmt, arg...)        do { if (isp_drv_DbgLogEnable_ERROR  ) { BASE_LOG_ERR(fmt, ##arg); } } while(0)
 #define LOG_AST(cond, fmt, arg...)  do { if (isp_drv_DbgLogEnable_ASSERT ) { BASE_LOG_AST(cond, fmt, ##arg); } } while(0)
+#else
+#define LOG_VRB(fmt, arg...)        
+#define LOG_DBG(fmt, arg...)        
+#define LOG_INF(fmt, arg...)        
+#define LOG_WRN(fmt, arg...)        
+#define LOG_ERR(fmt, arg...)        
+#define LOG_AST(cond, fmt, arg...)  
+#endif
 
 
 // for debug CQ virtual table
diff --git a/mediatek/platform/mt6595/kernel/core/camera_isp.c b/mediatek/platform/mt6595/kernel/core/camera_isp.c
index 13c8112..ad65496 100755
--- a/mediatek/platform/mt6595/kernel/core/camera_isp.c
+++ b/mediatek/platform/mt6595/kernel/core/camera_isp.c
@@ -71,7 +71,8 @@ typedef bool                    MBOOL;
 //#define LOG_WRN(fmt, arg...)    printk(KERN_ERR "[ISP][%s]Warning" fmt,__FUNCTION__, ##arg)
 //#define LOG_ERR(fmt, arg...)    printk(KERN_ERR   "[ISP][%s]Err(%5d):" fmt, __FUNCTION__,__LINE__, ##arg)
 
-
+#define ISP_LOG_SWITCH 0
+#if ISP_LOG_SWITCH
 #define LOG_VRB(format, args...)    xlog_printk(ANDROID_LOG_VERBOSE, "ISP", "[%s] " format, __FUNCTION__, ##args)
 #define LOG_DBG(format, args...)    xlog_printk(ANDROID_LOG_DEBUG  , "ISP", "[%s] " format, __FUNCTION__, ##args)
 //Both ANDROID_LOG_DEBUG and ANDROID_LOG_VERBOSE can be logged only to mobile log,
@@ -80,7 +81,14 @@ typedef bool                    MBOOL;
 #define LOG_WRN(format, args...)    xlog_printk(ANDROID_LOG_WARN   , "ISP", "[%s] WARNING: " format, __FUNCTION__, ##args)
 #define LOG_ERR(format, args...)    xlog_printk(ANDROID_LOG_ERROR  , "ISP", "[%s, line%04d] ERROR: " format, __FUNCTION__, __LINE__, ##args)
 #define LOG_AST(format, args...)    xlog_printk(ANDROID_LOG_ASSERT , "ISP", "[%s, line%04d] ASSERT: " format, __FUNCTION__, __LINE__, ##args)
-
+#else
+#define LOG_VRB(format, args...)
+#define LOG_DBG(format, args...)
+#define LOG_INF(format, args...)
+#define LOG_WRN(format, args...)
+#define LOG_ERR(format, args...)
+#define LOG_AST(format, args...)
+#endif
 /*******************************************************************************
 *
 ********************************************************************************/
@@ -823,7 +831,7 @@ static MUINT32 g_DmaErr_p1[nDMA_ERR] = {0};
         do_div( sec, 1000 );    \
         usec = do_div( sec, 1000000);\
 }
-#if 1
+#if 0
 #define IRQ_LOG_KEEPER(irq,ppb,logT,fmt,...) do{\
     char* ptr; \
     char* pDes;\
@@ -848,7 +856,7 @@ static MUINT32 g_DmaErr_p1[nDMA_ERR] = {0};
     }     \
 }while(0);
 #else
-#define IRQ_LOG_KEEPER(irq,ppb,logT,fmt,...)  xlog_printk(ANDROID_LOG_DEBUG  , "KEEPER", "[%s] " fmt, __FUNCTION__, ##__VA_ARGS__)
+#define IRQ_LOG_KEEPER(irq,ppb,logT,fmt,...)
 #endif
 
 #if 1
diff --git a/mediatek/platform/mt6595/kernel/core/include/mach/mt_freqhopping.h b/mediatek/platform/mt6595/kernel/core/include/mach/mt_freqhopping.h
index 85e7d50..326f33b 100755
--- a/mediatek/platform/mt6595/kernel/core/include/mach/mt_freqhopping.h
+++ b/mediatek/platform/mt6595/kernel/core/include/mach/mt_freqhopping.h
@@ -33,24 +33,8 @@
 
     #define FH_MSG_DEBUG   FH_MSG
 #else
-    #if 1 //log level is 6 xlog
-        #define FH_MSG(fmt, args...) \
-        do {    \
-        		xlog_printk(ANDROID_LOG_DEBUG, FHTAG, fmt, \
-                    	##args ); \
-        } while(0);
-    #else //log level is 4 (printk)
-        #define FH_MSG(fmt, args...) \
-        do {    \
-        		printk( FHTAG""fmt" \n", \
-                    	##args ); \
-        } while(0);
-    #endif
-
-    #define FH_MSG_DEBUG(fmt, args...)\
-        do {    \
-        		printk( FHTAG""fmt" \n", ##args ); \
-        } while(0);
+    #define FH_MSG(fmt, args...)
+    #define FH_MSG_DEBUG   FH_MSG
 #endif
 
 
diff --git a/mediatek/platform/mt6595/kernel/core/mt_clkmgr.c b/mediatek/platform/mt6595/kernel/core/mt_clkmgr.c
index 537b1af..8111321 100644
--- a/mediatek/platform/mt6595/kernel/core/mt_clkmgr.c
+++ b/mediatek/platform/mt6595/kernel/core/mt_clkmgr.c
@@ -37,7 +37,7 @@
 //#define DISP_CLK_LOG
 //#define SYS_LOG
 //#define MUX_LOG_TOP
-#define MUX_LOG
+//#define MUX_LOG
 //#define PLL_LOG_TOP
 //#define PLL_LOG
 
@@ -46,7 +46,7 @@
 /************************************************
  **********         log debug          **********
  ************************************************/
-
+#define clk_debug_switch 0 //1
 #define USING_XLOG
 
 #ifdef USING_XLOG 
@@ -75,12 +75,18 @@
 #define clk_warn(fmt, args...)      \
     printk(KERN_WARNING TAG);       \
     printk(KERN_CONT fmt, ##args)
+#if clk_debug_switch
 #define clk_info(fmt, args...)      \
     printk(KERN_NOTICE TAG);        \
     printk(KERN_CONT fmt, ##args)
 #define clk_dbg(fmt, args...)       \
     printk(KERN_INFO TAG);          \
     printk(KERN_CONT fmt, ##args)
+#else
+#define clk_info(fmt, args...)      
+#define clk_dbg(fmt, args...)       
+#endif
+
 #define clk_ver(fmt, args...)       \
     printk(KERN_DEBUG TAG);         \
     printk(KERN_CONT fmt, ##args)
diff --git a/mediatek/platform/mt6595/kernel/core/mt_cpufreq.c b/mediatek/platform/mt6595/kernel/core/mt_cpufreq.c
index 1d9c5d3..c1a4eee 100755
--- a/mediatek/platform/mt6595/kernel/core/mt_cpufreq.c
+++ b/mediatek/platform/mt6595/kernel/core/mt_cpufreq.c
@@ -103,49 +103,49 @@ extern void hp_limited_cpu_num(int num); // TODO: ask Marc to provide the head f
 
 /* for DVFS OPP table */ // TODO: necessary or just specify in opp table directly???
 
-#define DVFS_BIG_F0 (1898000) /* KHz */
-#define DVFS_BIG_F1 (1495000) /* KHz */
+#define DVFS_BIG_F0 (1897992) /* KHz */
+#define DVFS_BIG_F1 (1494996) /* KHz */
 #define DVFS_BIG_F2 (1365000) /* KHz */
 #define DVFS_BIG_F3 (1248000) /* KHz */
-#define DVFS_BIG_F4 (1144000) /* KHz */
-#define DVFS_BIG_F5 (1001000) /* KHz */
-#define DVFS_BIG_F6 (806000)  /* KHz */
-#define DVFS_BIG_F7 (403000)  /* KHz */
+#define DVFS_BIG_F4 (1143996) /* KHz */
+#define DVFS_BIG_F5 (1000992) /* KHz */
+#define DVFS_BIG_F6 (805992)  /* KHz */
+#define DVFS_BIG_F7 (402996)  /* KHz */
 
 #if defined(SLT_VMAX)
-#define DVFS_BIG_V0 (1150)    /* mV */
+#define DVFS_BIG_V0 (1250)    /* mV */
 #else
-#define DVFS_BIG_V0 (1100)    /* mV */
+#define DVFS_BIG_V0 (1200)    /* mV */
 #endif
-#define DVFS_BIG_V1 (1079)    /* mV */
-#define DVFS_BIG_V2 (1050)    /* mV */
-#define DVFS_BIG_V3 (1032)    /* mV */
-#define DVFS_BIG_V4 (1000)    /* mV */
-#define DVFS_BIG_V5 (963)     /* mV */
-#define DVFS_BIG_V6 (914)     /* mV */
-#define DVFS_BIG_V7 (814)     /* mV */
-
-#define DVFS_LITTLE_F0 (1690000) /* KHz */
-#define DVFS_LITTLE_F1 (1495000) /* KHz */
+#define DVFS_BIG_V1 (1179)    /* mV */
+#define DVFS_BIG_V2 (1150)    /* mV */
+#define DVFS_BIG_V3 (1132)    /* mV */
+#define DVFS_BIG_V4 (1100)    /* mV */
+#define DVFS_BIG_V5 (1063)     /* mV */
+#define DVFS_BIG_V6 (1014)     /* mV */
+#define DVFS_BIG_V7 (914)     /* mV */
+
+#define DVFS_LITTLE_F0 (1689996) /* KHz */
+#define DVFS_LITTLE_F1 (1494996) /* KHz */
 #define DVFS_LITTLE_F2 (1365000) /* KHz */
 #define DVFS_LITTLE_F3 (1248000) /* KHz */
-#define DVFS_LITTLE_F4 (1144000) /* KHz */
-#define DVFS_LITTLE_F5 (1001000) /* KHz */
-#define DVFS_LITTLE_F6 (806000)  /* KHz */
-#define DVFS_LITTLE_F7 (403000)  /* KHz */
+#define DVFS_LITTLE_F4 (1143996) /* KHz */
+#define DVFS_LITTLE_F5 (1000992) /* KHz */
+#define DVFS_LITTLE_F6 (805992)  /* KHz */
+#define DVFS_LITTLE_F7 (402996)  /* KHz */
 
 #if defined(SLT_VMAX)
-#define DVFS_LITTLE_V0 (1150)    /* mV */
+#define DVFS_LITTLE_V0 (1250)    /* mV */
 #else
-#define DVFS_LITTLE_V0 (1125)    /* mV */
+#define DVFS_LITTLE_V0 (1225)    /* mV */
 #endif
-#define DVFS_LITTLE_V1 (1079)    /* mV */
-#define DVFS_LITTLE_V2 (1050)    /* mV */
-#define DVFS_LITTLE_V3 (1023)    /* mV */
-#define DVFS_LITTLE_V4 (1000)    /* mV */
-#define DVFS_LITTLE_V5 (963)     /* mV */
-#define DVFS_LITTLE_V6 (914)     /* mV */
-#define DVFS_LITTLE_V7 (814)     /* mV */
+#define DVFS_LITTLE_V1 (1179)    /* mV */
+#define DVFS_LITTLE_V2 (1150)    /* mV */
+#define DVFS_LITTLE_V3 (1123)    /* mV */
+#define DVFS_LITTLE_V4 (1100)    /* mV */
+#define DVFS_LITTLE_V5 (1063)     /* mV */
+#define DVFS_LITTLE_V6 (1014)     /* mV */
+#define DVFS_LITTLE_V7 (914)     /* mV */
 
 #define PWR_THRO_MODE_LBAT_1365MHZ	BIT(0)
 #define PWR_THRO_MODE_BAT_OC_806MHZ	BIT(1)
@@ -162,6 +162,7 @@ extern void hp_limited_cpu_num(int num); // TODO: ask Marc to provide the head f
 #define HEX_FMT "0x%08x"
 #undef TAG
 
+#define cpufreq_debug_switch 0 //1
 #ifdef USING_XLOG
 #include <linux/xlog.h>
 
@@ -186,12 +187,18 @@ extern void hp_limited_cpu_num(int num); // TODO: ask Marc to provide the head f
 	printk(KERN_ERR TAG KERN_CONT fmt, ##args)
 #define cpufreq_warn(fmt, args...)      \
 	printk(KERN_WARNING TAG KERN_CONT fmt, ##args)
+#if cpufreq_debug_switch
 #define cpufreq_info(fmt, args...)      \
 	printk(KERN_NOTICE TAG KERN_CONT fmt, ##args)
 #define cpufreq_dbg(fmt, args...)       \
 	printk(KERN_INFO TAG KERN_CONT fmt, ##args)
 #define cpufreq_ver(fmt, args...)       \
 	printk(KERN_DEBUG TAG KERN_CONT fmt, ##args)
+#else
+#define cpufreq_info(fmt, args...)      
+#define cpufreq_dbg(fmt, args...)       
+#define cpufreq_ver(fmt, args...)      
+#endif
 
 #endif  /* USING_XLOG */
 
diff --git a/mediatek/platform/mt6595/kernel/core/mt_ptp.c b/mediatek/platform/mt6595/kernel/core/mt_ptp.c
index ac272c4..560ec91 100755
--- a/mediatek/platform/mt6595/kernel/core/mt_ptp.c
+++ b/mediatek/platform/mt6595/kernel/core/mt_ptp.c
@@ -2388,11 +2388,19 @@ static void ptp_init01_finish(struct ptp_det *det);
 #define ptp_notice(fmt, args...)    printk(KERN_NOTICE "[PTP] " fmt, ##args)
 #define ptp_info(fmt, args...)      printk(KERN_INFO "[PTP] " fmt, ##args)
 #define ptp_debug(fmt, args...)     printk(KERN_DEBUG "[PTP] " fmt, ##args)
-
+#define ptp_log_switch 0//1
 #if EN_ISR_LOG
+#if ptp_log_switch
 #define ptp_isr_info(fmt, args...)  ptp_notice(fmt, ##args)
 #else
+#define ptp_isr_info(fmt, args...)
+#endif
+#else
+#if ptp_log_switch
 #define ptp_isr_info(fmt, args...)  ptp_debug(fmt, ##args)
+#else
+#define ptp_isr_info(fmt, args...)
+#endif
 #endif
 
 #define FUNC_LV_MODULE          BIT(0)  /* module, platform driver interface */
diff --git a/mediatek/platform/mt6595/kernel/drivers/dispsys/ddp_color.c b/mediatek/platform/mt6595/kernel/drivers/dispsys/ddp_color.c
index 7c33459..63beb7c 100644
--- a/mediatek/platform/mt6595/kernel/drivers/dispsys/ddp_color.c
+++ b/mediatek/platform/mt6595/kernel/drivers/dispsys/ddp_color.c
@@ -805,7 +805,7 @@ SKY_TONE_H:
 
 };
 
-int color_dbg_en = 1;
+int color_dbg_en = 0;
 #define COLOR_ERR(fmt, arg...) printk(KERN_ERR "[COLOR] " fmt "\n", ##arg)
 #define COLOR_DBG(fmt, arg...) \
     do { if (color_dbg_en) printk(KERN_DEBUG "[COLOR] " fmt "\n", ##arg); } while (0)
@@ -1246,7 +1246,7 @@ static int _color_io(DISP_MODULE_ENUM module, int msg, unsigned long arg, void *
     DISPLAY_PQ_T * pq_index;
     DISPLAY_TDSHP_T * tdshp_index;
 
-    COLOR_ERR("_color_io: msg %x\n", msg);
+    COLOR_DBG("_color_io: msg %x\n", msg);
 
     switch (msg) {
         case DISP_IOCTL_SET_PQPARAM:
diff --git a/mediatek/platform/mt6595/kernel/drivers/m4u/m4u.c b/mediatek/platform/mt6595/kernel/drivers/m4u/m4u.c
index 6ae1f99..57d1980 100755
--- a/mediatek/platform/mt6595/kernel/drivers/m4u/m4u.c
+++ b/mediatek/platform/mt6595/kernel/drivers/m4u/m4u.c
@@ -819,8 +819,8 @@ int m4u_dealloc_mva(m4u_client_t *client, M4U_PORT_ID port, unsigned int mva)
     }
 	pMvaInfo->flags |= M4U_FLAGS_MVA_IN_FREE;
 	
-    M4ULOG_HIGH("m4u_dealloc_mva: larb=%d,module=%s,mva=0x%x, size=%d\n", 
-            m4u_port_2_larb_id(port), m4u_get_port_name(port), mva, pMvaInfo->size);
+    //M4ULOG_HIGH("m4u_dealloc_mva: larb=%d,module=%s,mva=0x%x, size=%d\n", 
+      //      m4u_port_2_larb_id(port), m4u_get_port_name(port), mva, pMvaInfo->size);
 
     if(!gM4U_debug_trigger_translation_fault)
     {
diff --git a/mediatek/platform/mt6595/kernel/drivers/m4u/m4u_debug.c b/mediatek/platform/mt6595/kernel/drivers/m4u/m4u_debug.c
index 2e84dc6..6ceaae2 100644
--- a/mediatek/platform/mt6595/kernel/drivers/m4u/m4u_debug.c
+++ b/mediatek/platform/mt6595/kernel/drivers/m4u/m4u_debug.c
@@ -9,7 +9,7 @@
 
 //global variables
 int gM4U_log_to_uart = 0;
-int gM4U_log_level = 2;
+int gM4U_log_level = 3;
 int gM4U_debug_trigger_translation_fault = 0;
 
 
diff --git a/mediatek/platform/mt6595/kernel/drivers/videox/disp_drv_log.h b/mediatek/platform/mt6595/kernel/drivers/videox/disp_drv_log.h
index f0563bf..e42e8b8 100644
--- a/mediatek/platform/mt6595/kernel/drivers/videox/disp_drv_log.h
+++ b/mediatek/platform/mt6595/kernel/drivers/videox/disp_drv_log.h
@@ -42,6 +42,10 @@ extern char dprec_error_log_buffer[];
 	
 #define DISPPR_HWOP(string, args...)  //dprec_logger_pr(DPREC_LOGGER_HWOP, string, ##args);
 #define DISPPR_ERROR(string, args...)  do{dprec_logger_pr(DPREC_LOGGER_ERROR, string, ##args);printk("[DISP][%s #%d]ERROR:"string,__func__, __LINE__, ##args); }while(0)
+#if 0
 #define DISPPR_FENCE(string, args...)  do{dprec_logger_pr(DPREC_LOGGER_FENCE, string, ##args);xlog_printk(ANDROID_LOG_DEBUG,   "fence/",  string, ##args); }while(0)
+#else
+#define DISPPR_FENCE(string, args...)
+#endif
 
 #endif // __DISP_DRV_LOG_H__
