diff --git a/system/netd/SoftapController.cpp b/system/netd/SoftapController.cpp
index f296980..e569c71 100644
--- a/system/netd/SoftapController.cpp
+++ b/system/netd/SoftapController.cpp
@@ -154,7 +154,7 @@ int SoftapController::setSoftap(int argc, char *argv[]) {
 #ifndef USE_MTK_WLAN
     asprintf(&wbuf, "interface=%s\ndriver=nl80211\nctrl_interface="
             "/data/misc/wifi/hostapd\nssid=%s\nchannel=%d\nieee80211n=1\n"
-            "hw_mode=g\nignore_broadcast_ssid=%d\n",
+            "hw_mode=a\nignore_broadcast_ssid=%d\n",
             argv[2], argv[3], channel, hidden);
 #else
 /*
@@ -172,7 +172,7 @@ int SoftapController::setSoftap(int argc, char *argv[]) {
     if (argc > 9) {
         asprintf(&wbuf, "interface=%s\ndriver=nl80211\nctrl_interface="
             "/data/misc/wifi/hostapd\nssid=%s\nchannel=%d\nieee80211n=1\n"
-            "hw_mode=g\nignore_broadcast_ssid=%d\nmax_num_sta=%d\n"
+            "hw_mode=a\nignore_broadcast_ssid=%d\nmax_num_sta=%d\n"
         #ifndef MTK_BSP_PACKAGE
             "eap_server=1\nwps_state=2\nconfig_methods=display physical_display push_button\n"
         #endif
@@ -182,7 +182,7 @@ int SoftapController::setSoftap(int argc, char *argv[]) {
     } else {
         asprintf(&wbuf, "interface=%s\ndriver=nl80211\nctrl_interface="
             "/data/misc/wifi/hostapd\nssid=%s\nchannel=%d\nieee80211n=1\n"
-            "hw_mode=g\nignore_broadcast_ssid=%d\n"
+            "hw_mode=a\nignore_broadcast_ssid=%d\n"
         #ifndef MTK_BSP_PACKAGE
             "eap_server=1\nwps_state=2\nconfig_methods=display physical_display push_button\n"
         #endif
diff --git a/system/netd/SoftapController.h b/system/netd/SoftapController.h
index 7063067..f7fd413 100644
--- a/system/netd/SoftapController.h
+++ b/system/netd/SoftapController.h
@@ -25,7 +25,7 @@
 #define AP_BSS_STOP_DELAY	500000
 #define AP_SET_CFG_DELAY	500000
 #define AP_DRIVER_START_DELAY	800000
-#define AP_CHANNEL_DEFAULT	6
+#define AP_CHANNEL_DEFAULT	40
 
 class SoftapController {
 public:
