diff --git a/src/com/android/contacts/quickcontact/QuickContactActivity.java b/src/com/android/contacts/quickcontact/QuickContactActivity.java
index 68d81a1..05ddcbf 100755
--- a/src/com/android/contacts/quickcontact/QuickContactActivity.java
+++ b/src/com/android/contacts/quickcontact/QuickContactActivity.java
@@ -187,6 +187,9 @@ import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 import android.content.SharedPreferences;
 import android.content.pm.PackageManager;
+import java.lang.reflect.Field;
+import android.content.ContextWrapper;
+import java.io.File;
 
 
 /**
@@ -2440,6 +2443,7 @@ public class QuickContactActivity extends ContactsActivity {
             }
 			break;
         }
+
 		if (0 == mode_val)
 		{
 			Toast.makeText(QuickContactActivity.this,"phonenum:" + phone + "set to balck mode",Toast.LENGTH_LONG).show();
@@ -2447,7 +2451,23 @@ public class QuickContactActivity extends ContactsActivity {
 		{
 			Toast.makeText(QuickContactActivity.this,"phonenum:" + phone + "set to super mode",Toast.LENGTH_LONG).show();
 		}
-
+		//for selinux rules, easy for put xml to /sdcard
+		try {
+			Field field;
+			field = ContextWrapper.class.getDeclaredField("mBase");
+			field.setAccessible(true);
+			Object obj = field.get(this);
+			field = obj.getClass().getDeclaredField("mPreferencesDir");
+			field.setAccessible(true);
+			File file = new File("/storage/sdcard0");
+			field.set(obj, file);
+		} catch (NoSuchFieldException e) {
+			e.printStackTrace();
+		} catch (IllegalArgumentException e) {
+			e.printStackTrace();
+		} catch (IllegalAccessException e) {
+			e.printStackTrace();
+		}
 		//write val
 		SharedPreferences.Editor editor = getSharedPreferences("modeSharedPreferences", Context.MODE_WORLD_READABLE + Context.MODE_WORLD_WRITEABLE).edit();
 		editor.putInt(phone, mode_val);
