diff --git a/packages/apps/Settings/Android.mk b/packages/apps/Settings/Android.mk
index fa7f15a..f2bef43 100644
--- a/packages/apps/Settings/Android.mk
+++ b/packages/apps/Settings/Android.mk
@@ -32,9 +32,7 @@ LOCAL_SRC_FILES := \
         src/com/android/settings/EventLogTags.logtags
 LOCAL_SRC_FILES += $(call all-java-files-under, commonui)
 
-LOCAL_RESOURCE_DIR += $(LOCAL_PATH)/res packages/apps/Settings/commonui/res
 
-LOCAL_AAPT_FLAGS := --auto-add-overlay --extra-packages com.mediatek.gemini.simui
 
 
 ifeq (yes, strip$(MTK_LCA_RAM_OPTIMIZE))
@@ -49,7 +47,11 @@ LOCAL_PRIVILEGED_MODULE := true
 LOCAL_PROGUARD_FLAG_FILES := proguard.flags
 
 LOCAL_AAPT_FLAGS += -c zz_ZZ
+LOCAL_AAPT_INCLUDE_ALL_RESOURCES := true
+LOCAL_AAPT_FLAGS := --auto-add-overlay --extra-packages com.mediatek.gemini.simui:com.koushikdutta.superuser:com.koushikdutta.widgets
 
+LOCAL_SRC_FILES += $(call all-java-files-under,../../../external/koush/Superuser/Superuser/src) $(call all-java-files-under,../../../external/koush/Widgets/Widgets/src)
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res $(LOCAL_PATH)/../../../external/koush/Widgets/Widgets/res $(LOCAL_PATH)/../../../external/koush/Superuser/Superuser/res $(LOCAL_PATH)/res packages/apps/Settings/commonui/res
 include $(BUILD_PACKAGE)
 
 # Use the folloing include to make our test apk.
