diff --git a/packages/apps/Settings/Android.mk b/packages/apps/Settings/Android.mk
index fa7f15a..50bd787 100644
--- a/packages/apps/Settings/Android.mk
+++ b/packages/apps/Settings/Android.mk
@@ -34,8 +34,10 @@ LOCAL_SRC_FILES += $(call all-java-files-under, commonui)
 
 LOCAL_RESOURCE_DIR += $(LOCAL_PATH)/res packages/apps/Settings/commonui/res
 
-LOCAL_AAPT_FLAGS := --auto-add-overlay --extra-packages com.mediatek.gemini.simui
+LOCAL_AAPT_FLAGS := --auto-add-overlay --extra-packages com.koushikdutta.superuser:com.koushikdutta.widgets:com.mediatek.gemini.simui
 
+LOCAL_SRC_FILES += $(call all-java-files-under,../../../external/koush/Superuser/Superuser/src) $(call all-java-files-under,../../../external/koush/Widgets/Widgets/src)
+LOCAL_RESOURCE_DIR += $(LOCAL_PATH)/../../../external/koush/Widgets/Widgets/res $(LOCAL_PATH)/../../../external/koush/Superuser/Superuser/res
 
 ifeq (yes, strip$(MTK_LCA_RAM_OPTIMIZE))
 LOCAL_AAPT_FLAGS += --utf16
@@ -47,7 +49,7 @@ LOCAL_CERTIFICATE := platform
 LOCAL_PRIVILEGED_MODULE := true
 
 LOCAL_PROGUARD_FLAG_FILES := proguard.flags
-
+LOCAL_AAPT_INCLUDE_ALL_RESOURCES := true
 LOCAL_AAPT_FLAGS += -c zz_ZZ
 
 include $(BUILD_PACKAGE)
