diff --git a/mediatek/custom/ckt95_lw_kk/cgen/cfgdefault/CFG_WIFI_Default.h b/mediatek/custom/ckt95_lw_kk/cgen/cfgdefault/CFG_WIFI_Default.h
index 09ca8e0..0b7e2e1 100755
--- a/mediatek/custom/ckt95_lw_kk/cgen/cfgdefault/CFG_WIFI_Default.h
+++ b/mediatek/custom/ckt95_lw_kk/cgen/cfgdefault/CFG_WIFI_Default.h
@@ -125,7 +125,7 @@ WIFI_CFG_PARAM_STRUCT stWifiCfgDefault =
       0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 }}, /* ucReverse5[72] */
 #endif	 
 	  0x01, // TX_PWR_PARAM_T is VALID
-	  0x01, // 5G band is supported
+	  0x00, // 5G band is supported
 	  0x01, // 2.4GHz band edge power enabled
 	  0x26, // cBandEdgeMaxPwrCCK
 	  0x1E, // cBandEdgeMaxPwrOFDM20
