diff --git a/goldsand/packages/apps/HiCam/src/com/hicam/gallery/filemanager/ImageCacheService.java b/goldsand/packages/apps/HiCam/src/com/hicam/gallery/filemanager/ImageCacheService.java
index 801fdf3..472620e 100755
--- a/goldsand/packages/apps/HiCam/src/com/hicam/gallery/filemanager/ImageCacheService.java
+++ b/goldsand/packages/apps/HiCam/src/com/hicam/gallery/filemanager/ImageCacheService.java
@@ -102,7 +102,7 @@ public class ImageCacheService {
             return;
         }
 //        String path = cacheDir.getAbsolutePath() + "/" + DIR_NAME;
-        String path = cacheDir.getAbsolutePath() + "/" + DIR_NAME;
+        String path = cacheDir.getAbsolutePath() + "/Android/" + DIR_NAME;
         mImageCacheDir = path;
         File imgCacheDir = new File(path);
         if(imgCacheDir.exists())
diff --git a/packages/apps/Settings/res/xml/mtk_usb_settings.xml b/packages/apps/Settings/res/xml/mtk_usb_settings.xml
index cf27b4c..a79ac5a 100755
--- a/packages/apps/Settings/res/xml/mtk_usb_settings.xml
+++ b/packages/apps/Settings/res/xml/mtk_usb_settings.xml
@@ -20,7 +20,7 @@
     <PreferenceCategory
             android:key="usb_category"
             android:title="@string/usb_connection_category" />
-
+<!--
     <com.mediatek.settings.deviceinfo.UsbPreference
         android:key="usb_mtp"
         android:title="@string/usb_mtp_title"
@@ -32,4 +32,5 @@
         android:title="@string/usb_ptp_title"
         android:summary="@string/usb_ptp_summary"
         />
+-->
 </PreferenceScreen>
diff --git a/packages/apps/Settings/src/com/mediatek/settings/deviceinfo/UsbSettingsExts.java b/packages/apps/Settings/src/com/mediatek/settings/deviceinfo/UsbSettingsExts.java
index 8cc0660..c820e33 100755
--- a/packages/apps/Settings/src/com/mediatek/settings/deviceinfo/UsbSettingsExts.java
+++ b/packages/apps/Settings/src/com/mediatek/settings/deviceinfo/UsbSettingsExts.java
@@ -73,6 +73,7 @@ public class UsbSettingsExts {
 		PreferenceScreen root = usbSettings.getPreferenceScreen();
 		if (root == null) return null;
 
+		/*
 		mMtp = (UsbPreference) root.findPreference(KEY_MTP);
 		mMtp.setOnPreferenceChangeListener(usbSettings);
 		mUsbPreferenceList.add(mMtp);
@@ -80,7 +81,7 @@ public class UsbSettingsExts {
 		mPtp = (UsbPreference) root.findPreference(KEY_PTP);
 		mPtp.setOnPreferenceChangeListener(usbSettings);
 		mUsbPreferenceList.add(mPtp);
-
+		*/
 		Context context = usbSettings.getActivity();
 		// Add UMS Mode
 		String umsConfig = android.os.SystemProperties.get(
@@ -114,7 +115,7 @@ public class UsbSettingsExts {
 
 		// Add BUild-in CD Mode
 		String bicrConfig = android.os.SystemProperties.get(PROPERTY_USB_BICR, FUNCTION_NOT_SUPPORT);
-		boolean bicrExist = bicrConfig.equals(FUNCTION_SUPPORT);
+		boolean bicrExist = false; //bicrConfig.equals(FUNCTION_SUPPORT);
 		if (bicrExist) {
 			PreferenceCategory bicrCategory = new PreferenceCategory(context);
 			bicrCategory.setTitle(R.string.usb_connect_as_cdrom_category);
