diff --git a/mediatek/config/mt6595/init.rc b/mediatek/config/mt6595/init.rc
index accf551..95a73f2 100755
--- a/mediatek/config/mt6595/init.rc
+++ b/mediatek/config/mt6595/init.rc
@@ -1678,10 +1678,10 @@ service agpsd /system/bin/mtk_agpsd
     group gps ccci inet sdcard_r sdcard_rw misc
 
 service mnld /system/xbin/mnld
-    class main
-    user gps
-    group nvram gps inet misc sdcard_rw sdcard_r media_rw
-    socket mnld stream 660 gps system
+	class main
+	user root
+	group nvram gps inet misc sdcard_rw sdcard_r media_rw
+	socket mnld stream 660 gps system
 
 service ipod /system/bin/ipod
     user root
diff --git a/mediatek/external/mnl/mnl_aosp/mnld/src/mnld.c b/mediatek/external/mnl/mnl_aosp/mnld/src/mnld.c
index 0826ac6..fa49cfd 100644
--- a/mediatek/external/mnl/mnl_aosp/mnld/src/mnld.c
+++ b/mediatek/external/mnl/mnl_aosp/mnld/src/mnld.c
@@ -1641,10 +1641,12 @@ linux_gps_init (void)
 	if(g_is_1Hz)
 	{
 		init_cfg.fix_interval = 1000;               // 1Hz update rate
+		MND_MSG("#####into 1HZ mode ...\n");
 	}
 	else 
 	{
 		init_cfg.fix_interval = 200;               // 5Hz update rate
+		MND_MSG("#####into 5HZ mode ...\n");
 	}
 
 
@@ -2062,7 +2064,7 @@ static int launch_daemon_thread(void)
 
     MND_MSG("chip_id is %s\n", chip_id);
 
-    g_is_1Hz = 1;
+    g_is_1Hz = 0;
 
     /* adjust priority when 5 Hz Mode */        
     policy = sched_getscheduler(0);
