diff --git a/modules/drivers/gnss/src/raw_stream.cpp b/modules/drivers/gnss/src/raw_stream.cpp
index 1433bcd..e35b087 100644
--- a/modules/drivers/gnss/src/raw_stream.cpp
+++ b/modules/drivers/gnss/src/raw_stream.cpp
@@ -19,6 +19,7 @@
 
 #include <ros/ros.h>
 #include <std_msgs/String.h>
+#include <execinfo.h>
 
 #include "gnss/stream.h"
 #include "gnss/utils.h"
@@ -92,7 +93,19 @@ Stream *create_stream(const config::Stream &sd) {
 
     case config::Stream::kNtrip:
       if (!sd.ntrip().has_address()) {
-        ROS_ERROR("ntrip def has no address field.");
+		void *buffer[100];
+		char **strings;
+
+		int tmp = backtrace(buffer, 100);
+		strings = backtrace_symbols(buffer, tmp);
+
+		for (int j = 0; j < tmp; j++) {
+			ROS_ERROR("%s\n,", strings[j]);
+		}
+
+		free(strings);
+		ROS_ERROR("%s:[%d]\n", __func__, __LINE__);
+        ROS_ERROR("ghjsdf ntrip def has no address field.");
         return nullptr;
       }
       if (!sd.ntrip().has_port()) {
