diff --git a/envsetup.sh b/envsetup.sh
index d80e95c..6a6ee1c 100644
--- a/envsetup.sh
+++ b/envsetup.sh
@@ -1074,7 +1074,8 @@ function gdbclient() {
     echo "Usage: gdbclient <pid|processname> [port number]"
     return -1
   fi
-  local DEVICE=$(adb_get_product_device)
+  #local DEVICE=$(adb_get_product_device)
+  local DEVICE=$TARGET_PRODUCT
 
   if [ -z "$DEVICE" ]; then
     echo "Error: Unable to get device name. Please check if device is connected and ANDROID_SERIAL is set."
