diff --git a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_common.h b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_common.h
index f6b3cd8..53212ef 100644
--- a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_common.h
+++ b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_common.h
@@ -128,7 +128,7 @@
  * This compile option is used for MTK platform only.
  * By default, this compile option is disabled.
  */
-//#define CONFIG_ENABLE_REPORT_KEY_WITH_COORDINATE
+#define CONFIG_ENABLE_REPORT_KEY_WITH_COORDINATE
 
 /*
  * Note.
@@ -176,7 +176,7 @@
  * If this compile option is not defined, the SW ID mechanism for updating firmware will be disabled.
  * By default, this compile option is disabled.
  */
-#define CONFIG_UPDATE_FIRMWARE_BY_SW_ID
+//#define CONFIG_UPDATE_FIRMWARE_BY_SW_ID
 
 /*
  * Note.
diff --git a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_mtk.c b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_mtk.c
index e999d18..bc93fb9 100644
--- a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_mtk.c
+++ b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_mtk.c
@@ -62,7 +62,7 @@
 /*=============================================================*/
 
 #define MSG_TP_IC_NAME "msg2xxx" //"msg21xxA" or "msg22xx" or "msg26xxM" /* Please define the mstar touch ic name based on the mutual-capacitive ic or self capacitive ic that you are using */
-#define I2C_BUS_ID   (0)       // i2c bus id : 0 or 1
+#define I2C_BUS_ID   (2)       // i2c bus id : 0 or 1
 
 #define TPD_OK (0)
 
@@ -197,7 +197,7 @@ static int tpd_local_init(void)
 #ifdef CONFIG_TP_HAVE_KEY
 #ifdef CONFIG_ENABLE_REPORT_KEY_WITH_COORDINATE     
     // initialize tpd button data
-    tpd_button_setting(4, g_TpVirtualKey, g_TpVirtualKeyDimLocal); //MAX_KEY_NUM
+    tpd_button_setting(2, g_TpVirtualKey, g_TpVirtualKeyDimLocal); //MAX_KEY_NUM
 #endif //CONFIG_ENABLE_REPORT_KEY_WITH_COORDINATE  
 #endif //CONFIG_TP_HAVE_KEY  
 
diff --git a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_platform_porting_layer.c b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_platform_porting_layer.c
index 3901045..d04c712 100644
--- a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_platform_porting_layer.c
+++ b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_platform_porting_layer.c
@@ -78,13 +78,13 @@ static int _gTpdFlag = 0;
 /*=============================================================*/
 
 #ifdef CONFIG_TP_HAVE_KEY
-const int g_TpVirtualKey[] = {TOUCH_KEY_MENU, TOUCH_KEY_HOME, TOUCH_KEY_BACK, TOUCH_KEY_SEARCH};
+const int g_TpVirtualKey[] = {TOUCH_KEY_MENU, TOUCH_KEY_BACK};
 
 #ifdef CONFIG_ENABLE_REPORT_KEY_WITH_COORDINATE
 #define BUTTON_W (100)
 #define BUTTON_H (100)
 
-const int g_TpVirtualKeyDimLocal[MAX_KEY_NUM][4] = {{BUTTON_W/2*1,TOUCH_SCREEN_Y_MAX+BUTTON_H/2,BUTTON_W,BUTTON_H},{BUTTON_W/2*3,TOUCH_SCREEN_Y_MAX+BUTTON_H/2,BUTTON_W,BUTTON_H},{BUTTON_W/2*5,TOUCH_SCREEN_Y_MAX+BUTTON_H/2,BUTTON_W,BUTTON_H},{BUTTON_W/2*7,TOUCH_SCREEN_Y_MAX+BUTTON_H/2,BUTTON_W,BUTTON_H}};
+const int g_TpVirtualKeyDimLocal[MAX_KEY_NUM][4] = {{250,900,100,40},{400,900,100,40}};
 #endif //CONFIG_ENABLE_REPORT_KEY_WITH_COORDINATE
 #endif //CONFIG_TP_HAVE_KEY
 
@@ -600,4 +600,4 @@ void DrvPlatformLyrSetIicDataRate(struct i2c_client *pClient, u32 nIicDataRate)
 #elif defined(CONFIG_TOUCH_DRIVER_RUN_ON_MTK_PLATFORM)
     pClient->timing = nIicDataRate/1000;
 #endif
-}
\ No newline at end of file
+}
diff --git a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_platform_porting_layer.h b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_platform_porting_layer.h
index b684e69..4128d33 100644
--- a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_platform_porting_layer.h
+++ b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_platform_porting_layer.h
@@ -122,11 +122,9 @@
 
 #ifdef CONFIG_TP_HAVE_KEY
 #define TOUCH_KEY_MENU    KEY_MENU 
-#define TOUCH_KEY_HOME    KEY_HOMEPAGE 
 #define TOUCH_KEY_BACK    KEY_BACK
-#define TOUCH_KEY_SEARCH  KEY_SEARCH
 
-#define MAX_KEY_NUM (4)
+#define MAX_KEY_NUM (2)
 #endif //CONFIG_TP_HAVE_KEY
 
 #endif
diff --git a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_self_fw_control.c b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_self_fw_control.c
index 0fbba0b..d92e5aa 100644
--- a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_self_fw_control.c
+++ b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v3/mstar_drv_self_fw_control.c
@@ -931,26 +931,16 @@ static s32 _DrvFwCtrlParsePacket(u8 *pPacket, u16 nLength, TouchInfo_t *pInfo)
                 pInfo->nTouchKeyCode = 0;
                 pInfo->nTouchKeyMode = 0;
 
-                if (pPacket[5] == 4) // TOUCH_KEY_HOME
-                {
-                    pInfo->tPoint[0].nX = g_TpVirtualKeyDimLocal[1][0];
-                    pInfo->tPoint[0].nY = g_TpVirtualKeyDimLocal[1][1];
-                }
-                else if (pPacket[5] == 1) // TOUCH_KEY_MENU
+                if (pPacket[5] == 1) // TOUCH_KEY_MENU
                 {
                     pInfo->tPoint[0].nX = g_TpVirtualKeyDimLocal[0][0];
                     pInfo->tPoint[0].nY = g_TpVirtualKeyDimLocal[0][1];
-                }           
+                }
                 else if (pPacket[5] == 2) // TOUCH_KEY_BACK
                 {
-                    pInfo->tPoint[0].nX = g_TpVirtualKeyDimLocal[2][0];
-                    pInfo->tPoint[0].nY = g_TpVirtualKeyDimLocal[2][1];
+                    pInfo->tPoint[0].nX = g_TpVirtualKeyDimLocal[1][0];
+                    pInfo->tPoint[0].nY = g_TpVirtualKeyDimLocal[1][1];
                 }           
-                else if (pPacket[5] == 8) // TOUCH_KEY_SEARCH 
-                {	
-                    pInfo->tPoint[0].nX = g_TpVirtualKeyDimLocal[3][0];
-                    pInfo->tPoint[0].nY = g_TpVirtualKeyDimLocal[3][1];
-                }
                 else
                 {
                     DBG("multi-key is pressed.\n");
@@ -1077,26 +1067,16 @@ static s32 _DrvFwCtrlParsePacket(u8 *pPacket, u16 nLength, TouchInfo_t *pInfo)
                 pInfo->nTouchKeyCode = 0;
                 pInfo->nTouchKeyMode = 0;
 
-                if (pPacket[8] == 4) // TOUCH_KEY_HOME
-                {
-                    pInfo->tPoint[0].nX = g_TpVirtualKeyDimLocal[1][0];
-                    pInfo->tPoint[0].nY = g_TpVirtualKeyDimLocal[1][1];
-                }
-                else if (pPacket[8] == 1) // TOUCH_KEY_MENU
+                if (pPacket[8] == 1) // TOUCH_KEY_MENU
                 {
                     pInfo->tPoint[0].nX = g_TpVirtualKeyDimLocal[0][0];
                     pInfo->tPoint[0].nY = g_TpVirtualKeyDimLocal[0][1];
-                }           
+                }
                 else if (pPacket[8] == 2) // TOUCH_KEY_BACK
                 {
-                    pInfo->tPoint[0].nX = g_TpVirtualKeyDimLocal[2][0];
-                    pInfo->tPoint[0].nY = g_TpVirtualKeyDimLocal[2][1];
+                    pInfo->tPoint[0].nX = g_TpVirtualKeyDimLocal[1][0];
+                    pInfo->tPoint[0].nY = g_TpVirtualKeyDimLocal[1][1];
                 }           
-                else if (pPacket[8] == 8) // TOUCH_KEY_SEARCH 
-                {	
-                    pInfo->tPoint[0].nX = g_TpVirtualKeyDimLocal[3][0];
-                    pInfo->tPoint[0].nY = g_TpVirtualKeyDimLocal[3][1];
-                }
                 else
                 {
                     DBG("multi-key is pressed.\n");
