diff --git a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v2/msg2133_driver_gesture.c b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v2/msg2133_driver_gesture.c
index 53f0315..afc8468 100644
--- a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v2/msg2133_driver_gesture.c
+++ b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v2/msg2133_driver_gesture.c
@@ -15,12 +15,13 @@
 #include <mach/mt_typedefs.h>
 #include <mach/mt_boot.h>
 #include "cust_gpio_usage.h"
+#include <linux/proc_fs.h>
+#include <linux/string.h>
 
 /*Open OR Close Debug Info*/
 /*Ctp Power Off In Sleep ? */
 //#define TPD_CLOSE_POWER_IN_SLEEP
 
- //#define MSG_GESTURE_FUNCTION
 extern int ektf2k_probed;
 extern int have_ektf2k_chip_on_board;
 #define APS_TAG                  "[MSG2133]:"
@@ -76,7 +77,7 @@ static struct i2c_client *i2c_client = NULL;
 
 
 #ifdef MSG_GESTURE_FUNCTION
-#define MSG_GESTURE_FUNCTION_NODE_PROC
+//#define MSG_GESTURE_FUNCTION_NODE_PROC
 #define CTP_GESTURE_FUNCTION_AUTHORITY_PROC 0777 
 
 
diff --git a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v2/tpd_custom_msg2133.h b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v2/tpd_custom_msg2133.h
index c1b764c..b7a12c7 100644
--- a/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v2/tpd_custom_msg2133.h
+++ b/mediatek/custom/common/kernel/touchpanel/msg2133_and_ektf2k_v2/tpd_custom_msg2133.h
@@ -42,6 +42,7 @@
 #define TPD_TYPE_CAPACITIVE
 #define TPD_I2C_NUMBER           2
 
+#define __TP_GESTURE_WAKEUP__
 #if defined(__TP_GESTURE_WAKEUP__)
 	#define MSG_GESTURE_FUNCTION
 #endif
@@ -131,14 +132,9 @@
 	#define KEY_S_D		(GESTRUE_KEY_X+110),GESTRUE_KEY_Y,KEY_WIDTH_HEIGHT
 	#define KEY_V_D		(GESTRUE_KEY_X+120),GESTRUE_KEY_Y,KEY_WIDTH_HEIGHT
 	#define KEY_Z_D		(GESTRUE_KEY_X+130),GESTRUE_KEY_Y,KEY_WIDTH_HEIGHT
-
-	#define TPD_KEY_COUNT       17    
-	#define TPD_KEYS                { KEY_MENU, KEY_HOME, KEY_BACK, KEY_UP, KEY_DOWN, KEY_LEFT, KEY_RIGHT, KEY_POWER,\
-					KEY_M, KEY_O,KEY_W, KEY_C, KEY_E, KEY_L, KEY_S, KEY_V, KEY_Z}
-	#define TPD_KEYS_DIM            {{100,900,20,40},{250,900,100,40},{400,900,100,40},{KEY_UP_D},{KEY_DOWN_D},{KEY_LEFT_D},\
-					{KEY_RIGHT_D},{KEY_DOULBE_D},{KEY_M_D},{KEY_O_D},{KEY_W_D},{KEY_C_D},{KEY_E_D},{KEY_L_D},\
-					{KEY_S_D},{KEY_V_D},{KEY_Z_D}}
-
+	#define TPD_KEY_COUNT           2
+	#define TPD_KEYS                	{ KEY_MENU, KEY_BACK}
+	#define TPD_KEYS_DIM            {{250,900,100,40},{400,900,100,40}}
 	#else
 	#define TPD_KEY_COUNT           2
 	#define TPD_KEYS                	{ KEY_MENU, KEY_BACK}
@@ -151,7 +147,7 @@
 const char MSG_2133_BIN[94*1024]=
 {
 //#include "hicam_mstar_fw_0902.i"
-#include "hicam_mstar_fw_0903.i"
+#include "hicam_mstar_fw_0904.i"
 };
 
 
