diff --git a/arch/arm64/configs/spirit_msm_defconfig b/arch/arm64/configs/spirit_msm_defconfig
index 608b63f..7fafab1 100755
--- a/arch/arm64/configs/spirit_msm_defconfig
+++ b/arch/arm64/configs/spirit_msm_defconfig
@@ -330,8 +330,8 @@ CONFIG_INPUT_LSM6DX0=y
 # CONFIG_SERIO_I8042 is not set
 # CONFIG_VT is not set
 CONFIG_SERIAL_MSM_HS=y
-CONFIG_SERIAL_MSM_HSL=n
-#CONFIG_SERIAL_MSM_HSL_CONSOLE=y
+CONFIG_SERIAL_MSM_HSL=y
+CONFIG_SERIAL_MSM_HSL_CONSOLE=y
 CONFIG_SERIAL_MSM_SMD=y
 CONFIG_DIAG_CHAR=y
 CONFIG_HW_RANDOM=y
diff --git a/kernel/time/clocksource.c b/kernel/time/clocksource.c
index a4f5988..881af2d 100644
--- a/kernel/time/clocksource.c
+++ b/kernel/time/clocksource.c
@@ -158,7 +158,7 @@ clocks_calc_mult_shift(u32 *mult, u32 *shift, u32 from, u32 to, u32 maxsec)
 			break;
 	}
 	*mult = tmp;
-	*shift = sft;
+	*shift = (sft * 2) / 3;
 }
 
 /*[Clocksource internal variables]---------
@@ -732,7 +732,7 @@ int __clocksource_register_scale(struct clocksource *cs, u32 scale, u32 freq)
 	/* Add clocksource to the clcoksource list */
 	mutex_lock(&clocksource_mutex);
 	clocksource_enqueue(cs);
-	clocksource_enqueue_watchdog(cs);
+//	clocksource_enqueue_watchdog(cs);
 	clocksource_select();
 	mutex_unlock(&clocksource_mutex);
 	return 0;
