diff --git a/mediatek/config/mt6595/init.rc b/mediatek/config/mt6595/init.rc
index b376833..66f236c 100755
--- a/mediatek/config/mt6595/init.rc
+++ b/mediatek/config/mt6595/init.rc
@@ -1307,6 +1307,9 @@ service atcid-daemon-e /system/bin/atcid
 on early_property:ro.build.type=eng
 start atci_service
 start atcid-daemon-e
+on early_property:ro.build.type=user
+start atci_service
+start atcid-daemon-e
 
 service audio-daemon /system/bin/audiocmdservice_atci
     class main
