diff --git a/mediatek/frameworks-ext/native/services/surfaceflinger/SurfaceFlingerWatchDog.cpp b/mediatek/frameworks-ext/native/services/surfaceflinger/SurfaceFlingerWatchDog.cpp
index 6745b39..868fa60 100644
--- a/mediatek/frameworks-ext/native/services/surfaceflinger/SurfaceFlingerWatchDog.cpp
+++ b/mediatek/frameworks-ext/native/services/surfaceflinger/SurfaceFlingerWatchDog.cpp
@@ -199,9 +199,17 @@ bool SFWatchDog::threadLoop() {
     char value[PROPERTY_VALUE_MAX];
     sprintf(value, "%lld", stopTime);
     uint32_t ret = property_set("service.sf.status", value);
+    char value_test[PROPERTY_VALUE_MAX];
+    property_get("my.stop.wd", value_test, "0");
+    int mytest = atoi(value_test);
     if (mUpdateSreen) {
-        XLOGV("[SF-WD] mUpdateSreen: %d", mUpdateSreen);
-        aee_ioctl_wdt_kick(WDT_SETBY_SF);
+        XLOGD("[SF-WD] mUpdateSreen: = %d, value_test = %s, mytest = %d", mUpdateSreen, value_test, mytest);
+		if (mytest != 0) {
+			XLOGD("MY KICK WD");
+			aee_ioctl_wdt_kick(WDT_SETBY_SF);
+		} else {
+			XLOGD("sorry my stop test");
+		}
         mUpdateSreen = 0;
     }
     //else {
