diff --git a/hardware/libhardware_legacy/wifi/wifi.c b/hardware/libhardware_legacy/wifi/wifi.c
index 9710d33..403eb77 100644
--- a/hardware/libhardware_legacy/wifi/wifi.c
+++ b/hardware/libhardware_legacy/wifi/wifi.c
@@ -1040,6 +1040,7 @@ int wifi_change_fw_path(const char *fwpath)
     int len;
     int fd;
     int ret = 0;
+	int retry_count = 4;
 
     ALOGD("enter -->%s fwpath=%s\n", __func__, fwpath);
     
@@ -1051,10 +1052,19 @@ int wifi_change_fw_path(const char *fwpath)
         return -1;
     }
     len = strlen(fwpath) + 1;
-    if (TEMP_FAILURE_RETRY(write(fd, fwpath, len)) != len) {
-        ALOGE("Failed to write wlan fw path param (%s)", strerror(errno));
-        ret = -1;
-    }
+	do {
+		if (TEMP_FAILURE_RETRY(write(fd, fwpath, len)) != len) {
+			ALOGE("Failed to write wlan fw path param (%s), retry_count = %d", strerror(errno), retry_count);
+			retry_count--;
+			usleep(100 * 1000);
+			ret = -1;
+		}
+		else {
+			ret = 0;
+			ALOGD("%s:[%d]wrtie fw path success after retry %d times\n", __func__, __LINE__, (4 - retry_count));
+			break;
+		}
+	} while (0 < retry_count);
     close(fd);
 
 #ifdef CONFIG_HOTSPOT_MGR_SUPPORT
